Feat: Scaffolding a new plugin named github-repo-scaffolding

daniel-hutao opened the pull request:
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

Scaffolding a new plugin named `github-repo-scaffolding`

## Key Points

- [x] Documentation (new or existing) is updated.

lfbdev reviewed (approved):
/lgtm

IronCore864 commented on cmd/kubeprometheus/main.go:
```diff
@@ -12,22 +12,22 @@ const NAME = "kube-prometheus"
 type Plugin string
 
-// Install implements the installation of some GitHub Actions workflows.
+// Install implements the installation of the kube-prometheus.
```
Incorrect comment; same below.

daniel-hutao replied:
this file is `cmd/kubeprometheus/main.go`

IronCore864 commented on cmd/reposcaffolding/github/main.go:
```diff
@@ -0,0 +1,40 @@
+)
+
+// NAME is the name of this DevStream plugin.
+const NAME = "repo-scaffolding-github"
```
As per our discussion, the name should be "github-repo-scaffolding".

After second thought, we should also include the name of the programming language as part of the plugin, because, for example, golang plugin needs to handle helm chart templates rendering, while python plugin doesn't.

daniel-hutao replied:
github-repo-scaffolding is right.
but for the language suffix, maybe we need to discuss it more.

IronCore864 commented on docs/repo-scaffolding-github_plugin.md:
```diff
@@ -0,0 +1,21 @@
+## 1 kube-prometheus Plugin
```
Doc name isn't correct.

IronCore864 commented on examples/config.yaml:
```diff
@@ -105,3 +105,9 @@ tools:
+    kind: repo-scaffolding-github
+    version: 0.0.1
+  options: null
+    # TODO(daniel-hutao): complete it
```
missing EOF

daniel-hutao commented:
I've fixed the name: repo-scaffolding-github -> github-repo-scaffolding #b1a35f612b89f2c58853c47b9f387f5c6f551f07,
And added the util/github by the way.

daniel-hutao commented:
I've change the name from github-repo-scaffolding -> github-repo-scaffolding-golang
